packages/at_persistence_secondary_server/lib/src/notification/at_notification_callback.dart

```diff
@@ -12,22 +12,19 @@ class AtNotificationCallback {
     return _singleton;
   }
 
+  @Deprecated('No longer in use. Replaced with notificationCallback')
   var callbackMethods = <NotificationType, List<NotificationFunction>>{};
 
+  Function? notificationCallback;
+
   /// Method to register callback function
   void registerNotificationCallback(
       NotificationType notificationType, Function callback) {
-    var nf = NotificationFunction();
-    nf.isValid = true;
-    nf.function = callback;
-    var functions = callbackMethods[notificationType];
-    functions ??= <NotificationFunction>[];
-    _removeUnregisteredFunctions(functions);
-    functions.add(nf);
-    callbackMethods[notificationType] = functions;
+    notificationCallback = callback;
   }
 
   /// Method to deregister callback function
+  @Deprecated('No longer in use')
   void unregisterNotificationCallback(
```

Comment:
how do we unregister callback after your refactored code?

Reply:
With the latest code changes, we fetch the list of connections for every notifications and write to the connections whose `isMonitor` is set to true. So, we will not have stale connections and does not require to unregisterNotificationCallback on closing the connection.